kse/src/org/kse/gui/actions/SignJarAction.java

```diff
@@ -85,43 +85,88 @@ protected void doAction() {
 				return;
 			}
 
+			// set the keystore state
 			KeyStore keyStore = currentState.getKeyStore();
+
+			// set the provider history
 			Provider provider = history.getExplicitProvider();
 
+			// set the private key
 			PrivateKey privateKey = (PrivateKey) keyStore.getKey(alias, password.toCharArray());
-			X509Certificate[] certs = X509CertUtil.orderX509CertChain(X509CertUtil.convertCertificates(keyStore
-					.getCertificateChain(alias)));
 
+			// set the certificate
+			X509Certificate[] certs = X509CertUtil
+					.orderX509CertChain(X509CertUtil.convertCertificates(keyStore.getCertificateChain(alias)));
+
+			// set the key pair type
 			KeyPairType keyPairType = KeyPairUtil.getKeyPairType(privateKey);
 
+			// set the signer
+			String signer = KSE.getFullApplicationName();
+
+			// get the jars, signatures, and time stamp
 			DSignJar dSignJar = new DSignJar(frame, privateKey, keyPairType, alias);
 			dSignJar.setLocationRelativeTo(frame);
 			dSignJar.setVisible(true);
 
+			// check if jar sign dialog was successful
+			if (!dSignJar.isSuccessful()) {
+				return;
+			}
+
 			SignatureType signatureType = dSignJar.getSignatureType();
 			String signatureName = dSignJar.getSignatureName();
-			File inputJarFile = dSignJar.getInputJar();
-			File outputJarFile = dSignJar.getOutputJar();
+			File[] inputJarFile = dSignJar.getInputJar();
+			List<File> outputJarFile = dSignJar.getOutputJar();
 			String tsaUrl = dSignJar.getTimestampingServerUrl();
+			DigestType digestType = dSignJar.getDigestType();
+
+			// start jar signing process
+			DSignJarSigning dSignJarSigning = new DSignJarSigning(frame, inputJarFile, outputJarFile, privateKey, certs,
+					signatureType, signatureName, signer, digestType, tsaUrl, provider);
+			dSignJarSigning.setLocationRelativeTo(frame);
+			dSignJarSigning.startDSignJarSigning();
+			dSignJarSigning.setVisible(true);
 
-			if (signatureType == null) {
+			// check if jar signing was successful
+			if (!dSignJarSigning.isSuccessful()) {
 				return;
 			}
 
-			String signer = KSE.getFullApplicationName();
-
-			DigestType digestType = dSignJar.getDigestType();
-
-			if (inputJarFile.equals(outputJarFile)) {
-				JarSigner.sign(inputJarFile, privateKey, certs, signatureType, signatureName, signer, digestType,
-						tsaUrl, provider);
+			// check if exceptions were caught during jar signing
+			if (dSignJarSigning.getFileExceptions().size() > 0) {
+				String strOK = res.getString("SignJarAction.ButtonOK.message");
+				String strView = res.getString("SignJarAction.ButtonView.message");
+				// set view and ok buttons
+				Object[] buttons = { strView, strOK };
+				// set file count
+				String fileCount = Integer.toString(inputJarFile.length);
+				// set exception count
+				String errorCount = Integer.toString(dSignJarSigning.getFileExceptions().size());
```

Comment:
Hello @kaikramer and @Colbix,

- Is it possible to change the type of `fileCount` and `errorCount` from `String` to `Integer`?
_(especially for improve i18n on resources files)_

Could you make a PR?
or would you like that I make one?